videomorph/converter/tasklist.py

```diff
@@ -179,10 +179,10 @@ def _running_task(self):
     def _add_task(self, task):
         """Add a video file to the list."""
         # Invalid metadata
-        try:
+        if 'duration' in task.video.format_info:
```

Comment:
you cant reply an exception raising another exception, this line hangs the soft with any miss-formatted video

Reply:
The problem here is that sometimes the 'duration' key is **in** the underlaying dict but is not a valid number and you can get either a TypeError or a ValueError. As I don't really want to catch three differet exceptions in the caller, I did it this way just to solve a bug at that time. I agree that this could be done in a better way, but I think it's not as easy as changing a *try* for an *if*, because that *if* can really hang the app if float() raises an exception.
I suggest to give this change its own PR later on and use this PR for the original purpose of having MacOSX support.
What do you think?

Reply:
The problem here is that sometimes, with some "magic" videos, the `duration` key is actually `in` the underlaying dictionary, but is not a valid number (believe me, it happens). That's why we catch `TypeError`, and `ValueError`. Also we don't want to catch three different exceptions in the caller, so we did it this way and this solves a bug we got some time ago.
I agree that there could be a better way to do this, but I think this may not be as easy as changing a `try` for an `if`, because that `if` can potentially hang the app if you try to use `float()` to convert something that can not be converted to `float`.
I'd suggest that we give this fix or improvement its own PR, and let this PR for its original purpose of having MacOSX support.
What do you think?

Reply:
The test videos I have hang with the try statement but run perfectly with the if. The behaviour is that in miss-formetted video it will say it doesnt find duration as a valid key (exception 1), then it will display both exceptions at the same time (also the raise statement) and finally the system hangs. With the if, there is no primary exception and the second exception is displayed smoothly. I will make this an own PR.

Reply:
```bash
  File "/usr/lib/python3.7/site-packages/videomorph-1.4-py3.7.egg/videomorph/converter/tasklist.py", line 184, in _add_task
    duration = float(task.video.format_info['duration'])
KeyError: 'duration'
zsh: abort (core dumped)  videomorph
```

Reply:
Ohh, I see. We're not catching `KeyError` here (That's new!!). What about trying to catch the `KeyError` and test with the same video again?
The common behavior is that VM inform the user that the video could not be added to the list, because of the metadata problems... I know that maybe that's not the ideal behavior, but it's what we have right now.